### PR TITLE
SVCPLAN-4649: numerous module updates/additions

### PR DESCRIPTION
**Pull request by jakerundall**
ncsa/profile_gpu: v1.0.0 -> v1.0.3 (dcgmi (monitoring) fixes)
ncsa/profile_lustre: v1.4.4 -> v2.0.0 (monitoring; BREAKING tuning change)
ncsa/profile_ondemand: add v1.0.3 (for OOD)
ncsa/profile_slurm: v1.6.2 -> v2.1.0 (BREAKING resource dependency mgmt change; add id_check (fka passwd_check))
ncsa/profile_user_environment: v0.1.4 -> v0.2.0 (class to run FS quota display script at login)
osc/openondemand: add 4.2.1 (for OOD)
puppet-alternatives: add 4.0.0 (for Lmod, Slurm)
puppet-archive: add 6.0.2 (for Slurm, Lmod, Apptainer, Jenkins, Telegraf)
puppet/logrotate: add 6.0.0 (OOD, Slurm, NHC)
puppet/yum: 5.1.0 -> 5.3.0 (on Magnus since Dec 7, 2022)
puppetlabs/vcsrepo: add 5.0.0 (for OOD)
srstevens/gcsv5: add v2.1.9 (Globus)
treydock/apptainer: add 2.0.0 (helpful to modify /etc/apptainer/apptainer.conf)
treydock/munge: add 4.1.0 (for Slurm)
treydock/nhc: add 5.0.0 (for Slurm)

### CODE REVIEW

**Comment by jakerundall** [comment]
Tested via control-pup01 and confirmed that r10k builds a working environment with no-op on representative nodes (CentOS 7, RHEL 7, RHEL 8.8, RHEL 8, RHEL 9.2, RHEL 9, Rocky 8).

The only change that is really relevant for nodes controlled by control-pup01 is the update of puppet/yum, which has been used on Magnus Dec 7, 2022.

The other changes have been tested on Magnus (and in some cases other clusters).

File: Puppetfile [hunk 1]
```diff
@@ -23,60 +23,70 @@ mod 'ncsa/profile_email', tag: 'v0.2.5', git: 'https://github.com/ncsa/puppet-pr
...
 mod 'puppetlabs/inifile', '4.4.0'
 mod 'puppetlabs/mailalias_core', '1.0.6'
 mod 'puppetlabs/mysql', '12.0.3'
 mod 'puppetlabs/stdlib', '6.3.0'
+mod 'puppetlabs/vcsrepo', '5.0.0'
 mod 'puppetlabs/xinetd', '3.3.0'
 mod 'richardc/datacat', '0.6.2'
 mod 'saz/sudo', '6.0.0'
 mod 'saz/timezone', '6.1.0'
 mod 'southalc/vault_secrets', '0.4.2'
```
Comment (billglick) [inline Puppetfile:84]: We no longer use vault, so this should be removed.

```suggestion
```

**Review by billglick — CHANGES_REQUESTED** [review]
Suggested we should remove the `vault_secrets` module while we're updating these.

File: Puppetfile [hunk 2]
```diff
@@ -23,60 +23,70 @@ mod 'ncsa/profile_email', tag: 'v0.2.5', git: 'https://github.com/ncsa/puppet-pr
...
+mod 'ncsa/profile_ondemand', tag: 'v1.0.3', git: 'https://github.com/ncsa/puppet-profile_ondemand.git'
 mod 'ncsa/profile_pam_access', tag: 'v0.0.6', git: 'https://github.com/ncsa/puppet-profile_pam_access'
 mod 'ncsa/profile_postgres_server', tag: 'v0.1.1', git: 'https://github.com/ncsa/puppet-profile_postgres_server'
 mod 'ncsa/profile_puppet_agent', tag: 'v0.1.4', git: 'https://github.com/ncsa/puppet-profile_puppet_agent'
 mod 'ncsa/profile_puppet_master', tag: 'v0.1.6', git: 'https://github.com/ncsa/puppet-profile_puppet_master'
 mod 'ncsa/profile_rsyslog', tag: 'v0.3.0', git: 'https://github.com/ncsa/puppet-profile_rsyslog'
 mod 'ncsa/profile_selinux', tag: 'v0.1.0', git: 'https://github.com/ncsa/puppet-profile_selinux'
 mod 'ncsa/profile_session_log', tag: 'v1.0.0', git: 'https://github.com/ncsa/puppet-profile_session_log.git'
-mod 'ncsa/profile_slurm', tag: 'v1.6.2', git: 'https://github.com/ncsa/puppet-profile_slurm.git'
+mod 'ncsa/profile_slurm', tag: 'v2.1.0', git: 'https://github.com/ncsa/puppet-profile_slurm.git'
```
Comment (billglick) [inline Puppetfile:47]: Any suggestions for calling out in the control repo that there is a breaking change related to this upgrade?

e.g. As I'm preparing to upgrade modules for the upcoming mForge PM, how do I notice that this module introduces a breaking change? Obviously I should go read about each module that has a significant change, but any ideas of a way we can call attention here in the control repo?

File: Puppetfile (hunk 1 above)
Comment (jakerundall) [reply Puppetfile]: Done!

File: Puppetfile (hunk 2 above)
Comment (billglick) [reply Puppetfile:47]: I guess it is really on the project to check for these sorts of things. If there are changes to the major or minor point version they should check. But if it is just a patch version iteration then it is generally acceptable.

Comment (jakerundall) [reply Puppetfile:47]: I guess that's why I put "BREAKING" in the commit messages lines for profile_lustre and profile_slurm.

But I could certainly add comments before the relevant lines in Puppetfile if that would be helpful.

And/or put an advisory earlier in the commit message, e.g., first or second line, that this commit contains breaking changes.

I can do whatever you think will work best.

Comment (billglick) [reply Puppetfile:47]: I'm not sure anything more should be done beyond what you already did.

https://wiki.ncsa.illinois.edu/display/PUPPET/Create+a+new+Puppet+Control+Repo#CreateanewPuppetControlRepo-MergeupdatesfromtheNCSAcontrolrepotemplate seems insufficient in helping guide someone how to vet which changes to pull from control-repo back to their project. But that is probably where some better guidance needs to live for looking for breaking changes.

This thread can be ignored for the pull request.

**Review by billglick — APPROVED** [review]
Looks good

**Pull request merged**